syntax = "proto3";

package hello;

service HelloService {
    rpc SayHello (HelloRequest) returns (HelloResponse);
}

message HelloRequest {
    string name = 1;
}

message HelloResponse {
    string greeting = 1;
}

message User {
    string name = 1;
    string contact = 2;
}

message CreditCard {
    string number = 1;
    string expirationDate = 2;
    string cvv = 3;
}

message VectorClock {
  repeated int32 vcArray = 1;
  double timestamp = 2;
}

message FraudDetectionRequest {
  User user = 1;
  CreditCard creditCard = 2;
  VectorClock vectorClock = 3;
}

message FraudDetectionResponse {
  bool is_fraudulent = 1;
}

service FraudDetectionService {
  rpc DetectFraud(FraudDetectionRequest) returns (FraudDetectionResponse);
}